### ISSUE TITLE
Promotion / create PR feature for GitHub Enterprise server

### ISSUE THREAD

**Issue by MarkusNeuron** [akuity/kargo#1355]
# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate feature request.
* [ ] I've pasted the output of `kargo version`, if applicable.
* [ ] I've pasted logs, if applicable.

# Proposed Feature
With 0.3.0 we can now create PR as part of the promotion.
This is the only remaining feature that blocks us to onboard Kargo.

# Motivation
Enterprise corps use Github-Enterprise installations ;)

# Suggested Implementation
Because of the similarities of Github- / Github Enterprise-APIs I hope its fairly simple to support also Github Enterprise servers.

**Comment by jessesuen** [comment]
Yes, I think support for this might be as simple as introducing logic to infer that the hostname is enterprise github, then calling go-githug with the [`WithEnterpriseURLs`](https://github.com/google/go-github/blob/5c8ee0baaeb2b0859a4f58638d78bc9cd46c6be8/github/github.go#L484) option:

```golang
NewClient(httpClient).WithEnterpriseURLs(baseURL, uploadURL) 
```

How I see the UX for this:
1. Today, if we detect if the word "github" is in the hostname, we assume that it is GitHub. I trust that this will always be a safe assumption. However, currently we don't properly configure the client to use the new hostname. So the only thing left to do is: if "github" is in the hostname, but the hostname is not exactly "github.com", then we must call `WithEnterpriseURLs(baseURL, uploadURL)`. 

2. Even with the above, there would be a case we wouldn't handle, which is: a user's github enterprise server might have a hostname like: `git.mycompany.com`. In this case, we couldn't/wouldn't assume it is github, and so we would need a system-wide setting to explicitly indicate `git.mycompany.com` is of type GitHub (and not BitBucket, etc...).

@MarkusNeuron option 1 should be quite easy to implement with a few lines of code, but not sure if hostname guessing would work for your use case, or if you would need an explicit system setting to indictate it.

**Comment by jessesuen** [comment]
> how can users provide a hint (through a new CRD attribute, probably) 

I think we were mostly saying the same thing. The difference in my proposal is for the hint to be done at a system level, as opposed to a CRD-level attribute.

**Comment by krancour** [comment]
> The difference in my proposal is for the hint to be done at a system level, as opposed to a CRD-level attribute.

Agree. In hindsight, the CRD attribute suggestion suggestion didn't make sense.

**Comment by MarkusNeuron** [comment]
Hi @krancour and @jessesuen, first let me thank you to your great response times and cool suggestions. Mid- / Log-term I see now way around to manually configure the git provider & corresponding URL in some way ( CRD, annotations, ...) to flexible support different git providers. 

As a quick hack the proposed integration... 

> but the hostname is not exactly "github.com", then we must call WithEnterpriseURLs(baseURL, uploadURL)

... would close our gap. Anyhow thanks for consideration.

**Comment by krancour** [comment]
This should have been closed by #2158 which addressed the need to set baseURL for GtiHub Enterprise clients at the same time that the same problem was being solved for self-hosted GitLab.
